db_stress_tool/db_stress_test_base.cc

```diff
@@ -3714,6 +3715,15 @@ void StressTest::Open(SharedState* shared, bool reopen) {
     fprintf(stderr, "open error: %s\n", s.ToString().c_str());
     exit(1);
   }
+
+  if (db_->GetLatestSequenceNumber() < shared->GetPersistedSeqno()) {
```

Comment:
do we need to check the value of `FLAGS_atomic_flush` to determine if we want to exit on this error? referring to https://github.com/facebook/rocksdb/issues/12152#issuecomment-1907127406

Reply:
Not sure if I get the question - I don't think the comment is suggesting the check is valid depending on the atomic flush or not. It's stating whether manual flush flushes the specified or all the CF depending on atomic flush and that does not affect the check. 

See below for examples why the check works for both cases:
1. atomic_flush = true, 
cf1: {key1@1, key3@3}
cf2: {key2@2}

Any flush whether implicit or manual flush cf1 and cf2 so largest flush seqno 3 <= recovered db seqno = 3

2. atomic_flush = false, we must then have disable_wal=0 by our db_crashtest constraint

cf1: {key1@1, key3@3}
cf2: {key2@2}

In the same WAL it will looks like
cf1, key1@1
cf2 key2@2
cf1, key3@3

Flush on either cf1 or cf2 will lead to largest flush seqno 3 or 2:
- This WAL will not be deleted since it contains unflushed data 
- WAL replay will lead to memtable contains all these three keys  so recovered db seqno = 3

> but there it is not needed because WAL ensures consistent recovery across column families.

It means that if there is an issue with recovering to key2@2, the db will recover to key1@1 so recovered db seqno = 1. But then we will have an earlier check before we get to this point https://github.com/facebook/rocksdb/blob/cf768a2f9e5451a99aa774d1ef2eb19681b8df6f/db/db_impl/db_impl_open.cc#L1688-L1689 to catch the case where SST contains newer data than WAL.

Reply:
> atomic_flush = false, we must then have disable_wal=0 by our db_crashtest constraint

Thanks for the explanation. I think this is the key point for why the check will work regardless of `atomic_flush`